homeassistant/components/openuv/config_flow.py

```diff
@@ -41,34 +54,37 @@ def __init__(self):
         """Handle the start of the config flow."""
         from pyopenuv.util import validate_api_key
 
-        errors = {}
-
-        if user_input is not None:
-            identifier = '{0}, {1}'.format(
-                user_input.get(CONF_LATITUDE, self.hass.config.latitude),
-                user_input.get(CONF_LONGITUDE, self.hass.config.longitude))
-
-            if identifier in configured_instances(self.hass):
-                errors['base'] = 'identifier_exists'
-            else:
-                websession = aiohttp_client.async_get_clientsession(self.hass)
-                api_key_validation = await validate_api_key(
-                    user_input[CONF_API_KEY], websession)
-                if api_key_validation:
-                    return self.async_create_entry(
-                        title=identifier,
-                        data=user_input,
-                    )
-                errors['base'] = 'invalid_api_key'
-
-        data_schema = OrderedDict()
-        data_schema[vol.Required(CONF_API_KEY)] = str
-        data_schema[vol.Optional(CONF_LATITUDE)] = cv.latitude
-        data_schema[vol.Optional(CONF_LONGITUDE)] = cv.longitude
-        data_schema[vol.Optional(CONF_ELEVATION)] = vol.Coerce(float)
-
-        return self.async_show_form(
-            step_id='user',
-            data_schema=vol.Schema(data_schema),
-            errors=errors,
-        )
+        if not user_input:
+            return await self._show_form()
+
+        latitude = user_input.get(CONF_LATITUDE)
+        if not latitude:
+            latitude = self.hass.config.latitude
+        longitude = user_input.get(CONF_LONGITUDE)
+        if not longitude:
+            longitude = self.hass.config.longitude
+        elevation = user_input.get(CONF_ELEVATION)
+        if not elevation:
+            elevation = self.hass.config.elevation
+
+        identifier = '{0}, {1}'.format(latitude, longitude)
+        if identifier in configured_instances(self.hass):
+            return await self._show_form({'base': 'identifier_exists'})
```

Comment:
Mark the specific entry in the form that is causing the error, by using the corresponding form key instead of `'base' `. Eg if it's the latitude that is bad, use `CONF_LATITUDE`.